modules/parsers/parser-hdbdd/src/main/java/com/sap/xsk/parser/hdbdd/custom/ReferenceResolvingListener.java

```diff
@@ -111,6 +116,13 @@ public void enterAssociationTarget(CdsParser.AssociationTargetContext ctx) {
           resolvedSymbol.getIdToken().getLine(), resolvedSymbol.getIdToken().getCharPositionInLine()));
     }
 
+    if (ctx.getParent().getChild(ManagedForeignKeysContext.class, 0) == null
+        && ctx.getParent().getChild(UnmanagedForeignKeyContext.class, 0) == null) {
+      List<EntityElementSymbol> associationKeys = ((EntitySymbol) resolvedSymbol).getKeys();
+      associationSymbol.setForeignKeys(associationKeys);
+      associationSymbol.setManaged(true);
```

Comment:
@GrozevJunior Is this correct? We saw that the columns in the foreign key objects weren't set properly (it was just the name of the entity repeated for each column). This fixed it but maybe it's not being set at the right place.

Reply:
@d-velev You have a point there! Then you can do it in 'public void exitAssociation(@NotNull CdsParser.AssociationContext ctx)' in this case the target will be resolved and you could take the association by the context and jsut add the foreign keys.